packages/mdc-dialog/foundation.js

```diff
@@ -0,0 +1,206 @@
+    return strings;
+  }
+
+  static get defaultAdapter() {
+    return {
+      hasClass: (/* className: string */) => {},
+      addClass: (/* className: string */) => {},
+      removeClass: (/* className: string */) => {},
+      addScrollLockClass: (/* className: string */) => {},
+      removeScrollLockClass: (/* className: string */) => {},
+      registerInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      deregisterInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      registerDialogInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      deregisterDialogInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      registerDocumentKeydownHandler: (/* handler: EventListener */) => {},
+      deregisterDocumentKeydownHandler: (/* handler: EventListener */) => {},
+      registerAcceptHandler: (/* handler: EventListener */) => {},
+      deregisterAcceptHandler: (/* handler: EventListener */) => {},
+      registerCancelHandler: (/* handler: EventListener */) => {},
+      deregisterCancelHandler: (/* handler: EventListener */) => {},
+      registerFocusTrappingHandler: (/* handler: EventListener */) => {},
+      deregisterFocusTrappingHandler: (/* handler: EventListener */) => {},
+      numFocusableElements: () => {/* number of focusable elements */},
+      resetDialogFocus: () => {/* sets focus on first element in dialog */},
+      setDefaultFocus: () => /* sets focus */ {},
+      getFocusableElements: (/* handler: EventListener */) => /* NodeList */ {},
+      saveElementTabState: (/* el: Element */) => {},
+      restoreElementTabState: (/* el: Element */) => {},
+      makeElementUntabbable: (/* el: Element */) => {},
+      setBackgroundAttr: (/* attr: String, val: Boolean */) => {},
+      setDialogAttr: (/* attr: String, val: Boolean */) => {},
+      acceptAction: () => {/* accept function */},
+      cancelAction: () => {/* cancel function */},
+      getFocusedElement: () => {},
+      setFocusedElement: () => {},
+    };
+  }
+
+  constructor(adapter) {
+    super(Object.assign(MDCDialogFoundation.defaultAdapter, adapter));
+
+    this.lastFocusedElement_;
+    this.currentFocusedElIndex_ = -1;
+    this.inert_ = true;
+    this.isOpen_ = false;
+    this.componentClickHandler_ = () => this.cancel();
+    this.dialogClickHandler_ = (evt) => evt.stopPropagation();
+    this.acceptHandler_ = () => this.accept();
+    this.cancelHandler_ = () => this.cancel();
+    this.focusHandler_ = () => this.setFocus_();
+    this.documentKeydownHandler_ = (evt) => {
+      if (evt.key && evt.key === 'Escape' || evt.keyCode === 27) {
+        this.cancel();
+      }
+    };
+  }
+
+  init() {
+    const {OPEN} = MDCDialogFoundation.cssClasses;
+
+    if (this.adapter_.hasClass(OPEN)) {
+      this.isOpen_ = true;
+    } else {
+      this.lockTab_();
+      this.isOpen_ = false;
+    }
+  }
+
+  destroy() {}
+
+  getLastFocusTarget() {
```

Comment:
If this is called `getLastFocusTarget`, would a better name for `lastFocusedElement` be `lastFocusTarget`?